### PR TITLE
Fix: Limit number of components displayed on translations overview page

### PR DESCRIPTION
**Pull request by osori**
Before submitting pull request, please ensure that:

- [x] Every commit has a message which describes it
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them
- [x] Any code changes come with test case
- [x] Any new functionality is covered by user documentation

Here is my fix to #4220, which does the following:
1. Limit the number of "Other components" displayed to 10, preferring untranslated ones
2. Template: Add a link to the full list of components

### CODE REVIEW

File: weblate/templates/translation.html [hunk 1]
```diff
@@ -185,6 +185,9 @@ <h4 class="panel-title">
 <div class="panel panel-default">
   <div class="panel-heading"><h4 class="panel-title">{% trans "Other components" %}</h4></div>
 {% include "snippets/list-objects.html" with objects=other_translations name_source="component_name" label=_("Component") in_panel=True %}
+  <div class="panel-footer">
+    <a href="{% url 'project-language' lang=object.language.code project=object.component.project.slug %}">See all components</a>
```
Comment (nijel) [inline weblate/templates/translation.html:189]: Please make the text localizable. Also in similar context we use "Browse all ...", so please make the text consistent.

Comment (osori) [reply weblate/templates/translation.html:189]: Oops, my bad. I just changed the text to be more consistent and wrapped it with trans tag. 
Thank you for reviewing!

File: weblate/trans/views/basic.py [hunk 2]
```diff
@@ -315,6 +315,12 @@ def show_translation(request, project, component, lang):
         if test_component.can_add_new_language(user):
             other_translations.append(GhostTranslation(test_component, obj.language))
 
+    # Limit the number of other components displayed to 10, preferring untranslated ones
+    other_translations = sorted(
+        other_translations,
```
Comment (osori) [inline weblate/trans/views/basic.py:320]: I'm not sure what `black` would change here... it seems fine to me. Any ideas why this happens?

File: weblate/trans/views/basic.py [hunk 3]
```diff
@@ -315,6 +315,12 @@ def show_translation(request, project, component, lang):
...
+        key=lambda t : t.stats.translated_percent
```
Comment (nijel) [inline weblate/trans/views/basic.py:321]: I think it complains about this:

```suggestion
        key=lambda t : t.stats.translated_percent,
```

The best approach is install pre-commit hooks which will format the code, see https://docs.weblate.org/en/latest/contributing/code.html#coding-standard

Comment (osori) [reply weblate/trans/views/basic.py]: This is my first time committing using pre-commit and wow, this is great! I should have read the contributor docs thoroughly. Thanks a lot for your help!

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**